src/app/helpers/error.interceptor.ts

```diff
@@ -0,0 +1,19 @@
+import { HttpInterceptor, HttpRequest, HttpHandler, HttpEvent } from '@angular/common/http';
+import { Observable, throwError } from 'rxjs';
+import { AuthService } from '../login/services/auth.service';
+import { catchError } from 'rxjs/operators';
+
+export class ErrorInterceptor implements HttpInterceptor {
+
+  constructor(private authService: AuthService) { }
+
+  intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    return next.handle(request).pipe(catchError(error => {
+      if (error.status === 401) {
+        this.authService.logout();
```

Comment:
So if the employee is logged in and tries to reach (for example) /payrollReports, does it mean that the user is logged out instead of redirected to dashboard?

Reply:
What are all the possible ways to get 401 response?
- wrong username/password for /oauth2/token endpoint (display error message)
- expired or wrong access token (logout and redirect to login screen)
- insufficient privileges (this should not happen in our app if we don't make such a request)